posts/2025-05-xx_crossing_manfred/README.md

```diff
@@ -0,0 +1,182 @@
+highly composable, robust error handling, and automatic transactional
+consistency, all within a single language and runtime environment. This opens
+the door to powerful new patterns of code reuse and composability directly on
+the blockchain.
+
+## Understanding `cross` and `crossing`
+
+While basic interrealm calls provide access to another realm's functions and
+data (read-only), the optional `cross` keyword and `crossing()` function
+modifier introduce control over the execution *context*.
+
+* **Calling without `cross` (Non-crossing call):** When you call a function in
+  another realm *without* using `cross`, you are essentially executing that
+  remote code within your *current* realm's context and memory space. Direct
+  modification of the remote realm's state is restricted. Think of it like
+  borrowing a library from another service but running it locally. Trying to
+  modify memory of another realm without `cross` is forbidden and will result in
+  a panic. For example, if Realm A calls a function in Realm B that attempts
+  `realmB.someState = "new value"`, this operation will fail if not done within
+  a `crossing()` call.
+
+    ```ascii
+     Non-crossing Call (Realm A calls Realm B's function):
+
+     [Realm A Context]------------+
+     |                            |
+     | Executes Realm B's code    |
+     | (within Realm A's context) |
+     |                            |
+     +----------------------------+
+                       |
+                       | Reads data from
+                       V
+                    [Realm B State] (Read-Only Access)
+    ```
+
+* **Calling with `cross` (Crossing call):** When you call a `crossing()`
+  function using the `cross(realmB.Function)(...)` syntax, you explicitly shift
+  the execution context *into* the target realm (Realm B). The function now
+  executes with Realm B's permissions, environment (`std.CurrentRealm()`
+  changes), and can directly modify Realm B's state. This is like truly entering
+  the other microservice's environment to perform an operation; Realm A is
+  allowed to modify Realm B's memory, but this modification happens within Realm
+  B's context.
+
+    ```ascii
+     Crossing Call (Realm A calls Realm B's crossing function):
+
+     [Realm A Context] --cross()--> [Realm B Context]-----------+
+                                    |                           |
+                                    | Executes Realm B's code   |
+                                    | (within Realm B's context)|
+                                    | Can modify Realm B State  |
+                                    |                           |
+     [Realm A State] <----return----+- std.CurrentRealm() is B -+
+                                    |                           |
+                                    +---------------------------+
+    ```
+
+A common question is: What if Realm A crosses into Realm B, and then Realm B
```

Comment:
Here is my common question: When calling another realm's crossing function, why does the calling realm have to call with `cross`? Shouldn't this be the default and be handled automatically by the GnoVM? In other words, what is an example where calling another realm's crossing function should *not* cross?